pumpkin/src/proxy/velocity.rs

```diff
@@ -60,11 +60,36 @@ pub fn receive_plugin_response(
             ));
             return;
         }
-        // TODO: no unwrap
-        let addr: SocketAddr = buf.get_string().unwrap().parse().unwrap();
+        // TODO: dont default to localhost
+        let addr: SocketAddr = buf.get_string().unwrap().parse()
+            .unwrap_or_else(|_| ([127, 0, 0, 1], 25565).into());
```

Comment:
you probably should get the current address then `BASIC_CONFIG.server_address

Reply:
that address is the default player address, it panics when you use localhost to connect to the server

Reply:
didn't come up with a better solution

Reply:
then you should be able to use `client.address`.
Seems Paper does that as well: https://github.com/PaperMC/Paper/blob/master/patches/server/0731-Add-Velocity-IP-Forwarding-Support.patch#L202

Reply:
I was infact wrong about it being localhost, the address we get from velocity just doesn't include a port

Reply:
> I was in fact wrong about it being localhost, the address we get from velocity just doesn't include a port edit: should we just default the port to 0?

looks like paper just grabs the current port from the players Socket Address, I think we should do that as well
![image](https://github.com/user-attachments/assets/addc8ba8-8020-4b4a-b1b3-c99419f27452)